geode-apis-compatible-with-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/set/SremDUnitTest.java

```diff
@@ -125,89 +121,143 @@ public void shouldDistributeDataAmongMultipleServers_givenMultipleClients_removi
+        (i) -> jedis.srem(key, members2.get(i))).run();
 
-    Set<String> results = jedis3.smembers(key);
+    Set<String> results = jedis.smembers(key);
 
     assertThat(results).isEmpty();
   }
 
+
   @Test
-  public void shouldDistributeDataAmongMultipleServers_givenMultipleClients_removingSameDataFromSameSetConcurrently() {
+  public void shouldDistributeDataAmongCluster_thenRemoveSameDataFromSameSetConcurrently() {
 
     String key = "key";
 
     List<String> members = makeMemberList(SET_SIZE, "member-");
 
-    jedis3.sadd(key, members.toArray(new String[] {}));
+    jedis.sadd(key, members.toArray(new String[] {}));
 
     new ConcurrentLoopingThreads(SET_SIZE,
-        (i) -> jedis1.srem(key, members.get(i)),
-        (i) -> jedis2.srem(key, members.get(i))).run();
+        (i) -> jedis.srem(key, members.get(i)),
+        (i) -> jedis.srem(key, members.get(i))).run();
 
-    Set<String> results = jedis3.smembers(key);
+    Set<String> results = jedis.smembers(key);
 
     assertThat(results).isEmpty();
   }
 
+
   @Test
-  public void shouldDistributeDataAmongMultipleServers_givenMultipleClients_removingFromDifferentSetsConcurrently() {
+  public void shouldDistributeDataAmongCluster_thenRemoveFromDifferentSetsConcurrently() {
 
     String key1 = "key1";
     String key2 = "key2";
 
     List<String> members1 = makeMemberList(SET_SIZE, "member1-");
     List<String> members2 = makeMemberList(SET_SIZE, "member2-");
 
-    jedis3.sadd(key1, members1.toArray(new String[] {}));
-    jedis3.sadd(key2, members2.toArray(new String[] {}));
+    jedis.sadd(key1, members1.toArray(new String[] {}));
+    jedis.sadd(key2, members2.toArray(new String[] {}));
 
     new ConcurrentLoopingThreads(SET_SIZE,
-        (i) -> jedis1.srem(key1, members1.get(i)),
-        (i) -> jedis2.srem(key2, members2.get(i))).run();
+        (i) -> jedis.srem(key1, members1.get(i)),
+        (i) -> jedis.srem(key2, members2.get(i))).run();
 
-    Set<String> results1 = jedis3.smembers(key1);
-    Set<String> results2 = jedis3.smembers(key2);
+    Set<String> results1 = jedis.smembers(key1);
+    Set<String> results2 = jedis.smembers(key2);
 
     assertThat(results1).isEmpty();
     assertThat(results2).isEmpty();
 
   }
 
+
+  @Test
+  public void shouldDistributeDataAmongCluster_givenMultipleClientsOnDifferentServer_removingSameDataFromSameSetConcurrently() {
+
+    int redisServerPort2 = clusterStartUp.getRedisPort(2);
+    JedisCluster jedis2 =
+        new JedisCluster(new HostAndPort(LOCAL_HOST, redisServerPort), redisServerPort2);
+
+    String key = "key";
+
+    List<String> members = makeMemberList(SET_SIZE, "member1-");
+    jedis.sadd(key, members.toArray(new String[] {}));
+
+    new ConcurrentLoopingThreads(SET_SIZE,
+        (i) -> jedis.srem(key, members.get(i)),
+        (i) -> jedis2.srem(key, members.get(i))).run();
+
+    Set<String> results = jedis.smembers(key);
+
+    assertThat(results).isEmpty();
+
+    jedis2.close();
+  }
```

Comment:
This is the same test as `shouldDistributeDataAmongCluster_thenRemoveSameDataFromSameSetConcurrently`. There is no difference in using a different `JedisCluster` instance inside the concurrent loop. Even with the same `jedis` instance, each thread will create a separate connection.